### PR TITLE
Add package "rabbitmq-c"

### PR DESCRIPTION
**Pull request by Colysia**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/r/rabbitmq-c/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    set_urls("https://github.com/alanxz/rabbitmq-c/archive/refs/tags/$(version).zip",
+             "https://github.com/alanxz/rabbitmq-c.git")
+
+    add_versions("v0.11.0", "fad876075cbcf9a2f3dff27be66391f69117d5068529ba1d53e67fe59c2d88e1")
+
+    if is_plat("windows") then
+        add_configs("shared", {description = "Build shared library.", default = false, type = "boolean"})
+    elseif is_plat("macosx", "linux") then
+        add_deps("autoconf", "automake", "pkg-config")
```
Comment (waruqi) [inline packages/r/rabbitmq-c/xmake.lua:14]: why? it use cmake in on_install()

Comment (waruqi) [reply packages/r/rabbitmq-c/xmake.lua:14]: you should add `add_deps("cmake")` for all platfroms

File: packages/r/rabbitmq-c/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    end
+    add_configs("tools", {description = "Build the command line tools.", default = false, type = "boolean"})
+    add_configs("ssl", {description = "Build rabbitmq-c with SSL support.", default = false, type = "boolean"})
```
Comment (waruqi) [inline packages/r/rabbitmq-c/xmake.lua:17]: you need add `package:add("deps", "openssl")` if `package:config("ssl")` is enabled in on_load()

File: packages/r/rabbitmq-c/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        table.insert(configs, "-DBUILD_STATIC_LIBS=" .. (package:config("shared") and "OFF" or "ON"))
+        table.insert(configs, "-DBUILD_TOOLS=" .. (package:config("tools") and "ON" or "OFF"))
+        table.insert(configs, "-DENABLE_SSL_SUPPORT=" .. (package:config("ssl") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs)
+    end)
+
+    on_load(function (package)
+        if package:config("ssl") then
+            package:add("deps", "openssl")
+        end
```
Comment (waruqi) [inline packages/r/rabbitmq-c/xmake.lua:31]:
```lua
    if package:config("tools") then
        package:addenv("PATH", "bin")
    end
```

File: packages/r/rabbitmq-c/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    set_homepage("https://github.com/alanxz/rabbitmq-c")
+    set_description("This is a C-language AMQP client library for use with v2.0+ of the RabbitMQ broker.")
+
+    set_urls("https://github.com/alanxz/rabbitmq-c/archive/refs/tags/$(version).zip",
+             "https://github.com/alanxz/rabbitmq-c.git")
+
+    add_versions("v0.11.0", "fad876075cbcf9a2f3dff27be66391f69117d5068529ba1d53e67fe59c2d88e1")
+
+    add_deps("cmake")
+    add_configs("shared", {description = "Build shared library.", default = false, type = "boolean"})
```
Comment (waruqi) [inline packages/r/rabbitmq-c/xmake.lua:12]: why? it only support static library?

Comment (Colysia) [reply packages/r/rabbitmq-c/xmake.lua:12]: Sorry! I mistake the use of this config. There is still some problem, I close this pr for the moment.

**Pull request closed without merging**